services/harmony/app/frontends/ogc-edr/get-data-for-trajectory.ts

```diff
@@ -0,0 +1,280 @@
+import { keysToLowerCase } from '../../util/object';
+import { ParameterParseError, mergeParameters, parseWkt, validateWkt } from '../../util/parameter-parsing-helpers';
+import { Response, NextFunction } from 'express';
+import HarmonyRequest from '../../models/harmony-request';
+import { ServerError, RequestValidationError } from '../../util/errors';
+import { getDataCommon } from './get-data-common';
+
+// LINESTRING to POLYGON conversion side length, 0.0001 is about 11 meters in precision
+const LINESTRING_PRECISION = 0.0001;
+
+/**
+ * Calculates the signed area of a polygon given its vertices.
+ * If the area is positive, the points are in counter-clockwise order.
+ * If negative, the points are in clockwise order.
+ *
+ * @param points - The vertices of the polygon.
+ * @returns The signed area of the polygon.
+ */
+function calculateSignedArea(points: Array<{ x: number; y: number }>): number {
+  let area = 0;
+  const n = points.length;
+
+  for (let i = 0; i < n; i++) {
+    const p1 = points[i];
+    const p2 = points[(i + 1) % n]; // Wrap around to the first point
+    area += (p1.x * p2.y) - (p2.x * p1.y);
+  }
+
+  return area / 2;
+}
+
+/**
+* Ensures the points of a polygon are ordered counter-clockwise.
+*
+* @param points - The vertices of the polygon.
+* @returns The vertices in counter-clockwise order.
+*/
+function ensureCounterClockwise(points: Array<{ x: number; y: number }>): Array<{ x: number; y: number }> {
+  const area = calculateSignedArea(points);
+
+  // If the area is negative, the points are in clockwise order, so reverse them
+  if (area < 0) {
+    return points.reverse();
+  }
+
+  return points;
+}
+
+/**
+ * Converts a two-point segment (line) into a polygon with a buffer.
+ * Ensures that the polygon points are in counter-clockwise order.
+ *
+ * @param segment - The two points defining the line segment.
+ * @param sideLength - The buffer distance (side length) to create around the segment.
+ * @returns The vertices of the polygon in counter-clockwise order.
+ */
+function convertSegmentToPolygon(segment: Array<{ x: number; y: number }>, sideLength: number): Array<{ x: number; y: number }> {
```

Comment:
You use this type definition in a lot of place. It would probably be good to create a formal definition for it like
```
type PointArray = Array<{x: number, y: number}>;
```
Then this function declaration becomes
```
function convertSegmentToPolygon(segment: PointArray, sideLength: number): PointArray
```
Then you can change all the other functions that use this type to use your new type declaration as well.

Reply:
Or maybe better, just define a Point type, like this
```
type Point =  {x: number, y: number};
```
Then your function becomes this
```
function convertSegmentToPolygon(segment: Point[], sideLength: number): Point[]
```

Reply:
Actually, to enforce the two-point nature of a segment you could do this
```
function convertSegmentToPolygon(segment: [Point, Point], sideLength: number): Point[]
```

Reply:
Updated.